caramel/module/scripts/caramel.handlebars.js

```diff
@@ -513,14 +514,54 @@ engine('handlebars', (function () {
      */
     theme = function (page, contexts, js, css, code) {
         var file, template, path, area, blocks, helper, length, i, o, areas, block,
-            areaContexts, data, areaData, find, blockData, resolve, partials, findPartials, analyzePartials,
+            areaContexts, data, areaData, find, blockData, resolve, partials, findPartials, analyzePartials, testFunction,
             theme = caramel.theme(),
             meta = caramel.meta(),
-            xcd = meta.request.getHeader(caramelData);
+            xcd = meta.request.getHeader(caramelData),
+            cct = meta.request.getHeader(CaramelCompiledTemplates);
+
         js = js || [];
         css = css || [];
         code = code || [];
 
+        /**
+         * Set the Caramel-Compiled header to compile the required partial from the backend
+         * and send the HTML block to the frontend.
+         */
+        if (cct) {
+            areas = parse(cct);
+            if (areas != null) {
```

Comment:
Shall we do the area check here:

areaKeys = Object.keys(areas || {} ) [0];
if(!areas  && areas[areaKey]) {

We can then be always sure that path will point to a valid file name and not undefined.hbs